packages/rxjs-errors/src/filter-with-error.operator.ts

```diff
@@ -0,0 +1,25 @@
+import { of, throwError } from 'rxjs';
+import { concatMap } from 'rxjs/operators';
+
+import { RxjsFilterError } from './rxjs-filter-error';
+
+import type { OperatorFunction } from 'rxjs';
+
+export type FilterWithErrorOperator = <TInput, TOutput extends TInput = TInput>(
+    passingCondition: ((val: TInput) => boolean) | ((val: TInput) => val is TOutput),
+    errorCodeOrMsgFn: string | ((val: TInput) => string),
```

Comment:
errorCodeOrMsgFn -> errorStringOrMsgFn we are not describing anything related to codes yet